nuxeo-diff-pictures/src/main/java/org/nuxeo/diff/pictures/DiffPicturesServlet.java

```diff
@@ -97,74 +98,84 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             repo = Framework.getLocalService(RepositoryManager.class).getDefaultRepository().getName();
         }
 
-        // This try-with-resources does an implicit close() at the end
-        try (CoreSession coreSession = CoreInstance.openCoreSession(repo)) {
+        TransactionHelper.startTransaction();
+        boolean completedAbruptly = true;
```

Comment:
You could use `TransactionHelper.runInTransaction` to avoid doing yourself all the start/commit/completedAbruptly stuff.